### PR TITLE
Add optional function call to optional chaining

### PR DESCRIPTION
**Pull request by ilogico**
I noticed the syntax for the optional function call wasn't being tested `f?.()`.
I also slightly changed the test for optional method call to better guarantee `this` is correct.

### CODE REVIEW

**Review by ljharb — APPROVED** [review]
Pending confirmation that all the results here have been manually verified.

**Comment by ilogico** [comment]
I didn't confirm the results manually, I assumed they were the same as the rest.
I did confirm them on the latest Firefox, Chrome and TypeScript.

**Comment by ljharb** [comment]
In that case, please remove all the results you didn't confirm manually :-) future PRs can add the results as they're tested.

**Comment by ilogico** [comment]
I removed the results I couldn't confirmed.
I kept the false results which are guaranteed to be false (ie11 and firefox < 73)

**Review by chicoxyzzy — APPROVED** [review]
Both optional method call and optional function call tests return `true` in Safari 13.1 Beta and Firefox 74 Beta. Though we can add these results in the following PR.

File: data-es2016plus.js [hunk 1]
```diff
@@ -3874,23 +3874,38 @@ exports.tests = [
...
         res : {
-          babel7corejs2: true,
           typescript3_7corejs3: true,
           ie11: false,
           firefox10: false,
           firefox52: false,
           firefox73: false,
-          firefox74: true,
           chrome77: false,
-          chrome78: {val: 'flagged', note_id: "chrome-optional-chaining"},
```
Comment (chicoxyzzy) [inline data-es2016plus.js]: I think that this result is still valid, we should re-check it

File: data-es2016plus.js [hunk 2]
```diff
@@ -3874,23 +3874,38 @@ exports.tests = [
       {
         name: 'optional method call',
         exec: function(){/*
-          var foo = { baz: function () { return 42; } };
+          var foo = { baz: function () { return this.value; }, value: 42 };
           var bar = null;
           return foo?.baz() === 42 && bar?.baz() === undefined;
         */},
         res : {
-          babel7corejs2: true,
```
Comment (chicoxyzzy) [inline data-es2016plus.js]: This result is still valid

**Review by chicoxyzzy — CHANGES_REQUESTED** [review]
Thinking again, all removed test results from optional method call tests should be added again. I've rechecked them all manually.

**Comment by ilogico** [comment]
Should I add them again?

**Comment by chicoxyzzy** [comment]
@ilogico yes please

**Comment by ilogico** [comment]
Done

**Comment by chicoxyzzy** [comment]
Oh, sorry, I mentioned only optional method call test results, not optional function call. But that's ok, I'll check them too. Thanks!

**Review by chicoxyzzy — APPROVED** [review]
Optional function call results are also correct. Thank you again!

**Pull request merged**